lib/ansiblelint/errors.py

```diff
@@ -27,3 +34,19 @@ def __repr__(self):
 
         return formatstr.format(_id, self.message,
                                 self.filename, self.linenumber, self.line)
+
+    @property
+    def _hash_key(self):
+        return self.filename, self.linenumber, self.message, getattr(self.rule, 'id', 0)
+
+    def __lt__(self, other):
+        """Enable sorting of MatchError instances."""
```

Comment:
This docstring doesn't explain what the method does, it only refers to a side-effect and therefore is incorrect.
```suggestion
        """Return whether the current object is less than the other."""
```